modules/builtin/src/backend/variables/date.ts

```diff
@@ -15,6 +15,26 @@ class BoxedDate extends BaseVariable<BoxedDateType, DateConfig> {
     super(args)
   }
 
+  parse(text: string): BoxedDateType {
+    return moment(text).toDate()
+  }
+
+  equals({ other }: { other: Date }) {
+    return (
+      moment(this.value)
+        .toDate()
+        .getTime() === other.getTime()
+    )
+  }
+
+  isBefore({ other }: { other: Date }) {
+    return moment(this.value).toDate() < other
+  }
+
+  isAfter({ other }: { other: Date }) {
```

Comment:
Why use an object instead of comparing the value directly? Don't forget that users will be able to use this in the code editor. This would feel weird:

`myDate.isBefore({ other: myOtherDate})`

Reply:
I made it like that so that all operators have the same call signature. So you could do something like this
`var[operatorName](args)`. I thought it would be beneficial when generating the code but I haven't really seen any benefits so far so I can change it

Reply:
I hadn't considered that users could use these functions.

With that in mind the parses should take place inside the operations, and the arguments shouldn't be a single object because writing this : `myData.isBefore(myOtherDate)` is going to be a lot nicer than writing `myDate.isBefore({ other : myDate.parse(myOtherDate) }`